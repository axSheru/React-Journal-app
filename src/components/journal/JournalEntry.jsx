import React from 'react';

export const JournalEntry = () => {
    return (
        <div className='journal__entry pointer'>
            <div
                className="journal___entry-picture"
                style={{
                    backgroundSize: 'cover',
                    backgroundImage: 'url(https://www.tonica.la/__export/1588444612791/sites/debate/img/2020/05/02/the_mandalorian_portada.jpg_1902800913.jpg)',
                }}
            >
            </div>
            <div className="journal__entry-body">
                <p className='journal__entry-title'>
                    Un nuevo dÃ­a.
                </p>
                <p className='journal__entry-content'>
                    Deserunt aliquip ut reprehenderit sint fugiat.
                </p>
            </div>
            <div className="journal__entry-date-box">
                <span>Tuesday</span>
                <h4>22</h4>
            </div>
        </div>
    );
}
