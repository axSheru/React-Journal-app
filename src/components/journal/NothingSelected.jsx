import React from 'react'

export const NothingSelected = () => {
    return (
        <div className='nothing__main-content'>
            <p>
                Select an entry
                <br />
                or create a new one!
            </p>
            <i className='far fa-star fa-4x mt-5'></i>
        </div>
    );
}
