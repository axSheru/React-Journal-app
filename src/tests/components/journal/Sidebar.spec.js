import configureStore from 'redux-mock-store';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { Sidebar } from "../../../components/journal/Sidebar";

jest.mock( '../../../actions/auth', ()  =>{
    startLogout: jest.fn()
});

jest.mock( '../../../actions/notes', ()  =>{
    startNewNote: jest.fn()
});

describe('Pruebas en el componente <Sidebar />.', () => {

    const middlewares = [ thunk ];
    const mockStore = configureStore( middlewares );

    const initState = {
        auth: {
            uid: '1235',
            name: 'Alex'
        },
        notes: {
            notes: []
        }
    };

    let store = mockStore( initState );

    store.dispatch = jest.fn();

    const wrapper = mount(
        <Provider store={ store } >
            <Sidebar />
        </Provider>
    );

    test('Debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    /* test('Debe de disparar la acción startLogout.', () => {

        wrapper.find( 'button' ).prop( 'onClick' )();

        expect( startLogout ).toHaveBeenCalled();

    });

    test('Debe de disparar la acción startNewNote.', () => {
        
        wrapper.find( '.journal___new-entry' ).prop( 'onClick' )();

        expect( startNewNote ).toHaveBeenCalled();

    }); */

});