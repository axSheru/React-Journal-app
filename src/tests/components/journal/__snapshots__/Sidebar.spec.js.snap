// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Sidebar />. Debe de hacer match con el snapshot. 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="fa fa-moon"
      />
      <span>
         
        Alex
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal___new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
